AllReadyApp/Web-App/AllReady/Controllers/ActivityApiController.cs

```diff
@@ -140,34 +142,25 @@ public ActionResult GetCheckin(int id)
             }
         }
 
-        [HttpPost("{id}/signup")]
-        [Authorize] 
         [ValidateAntiForgeryToken]
-        public async Task<IActionResult> RegisterActivity(int id)
+        [HttpPost("signup")]
+        [Authorize]
+        public object RegisterActivity(ActivitySignupViewModel signupModel)
         {
-            var user = _allReadyDataAccess.GetUser(User.GetUserId());
-
-            var activity = _allReadyDataAccess.GetActivity(id);
-
-            if (activity == null)
+            if (signupModel == null)
             {
-                return HttpNotFound();
+                return HttpBadRequest();
             }
 
-            if (activity.UsersSignedUp == null)
+            if (!ModelState.IsValid)
             {
-                activity.UsersSignedUp = new List<ActivitySignup>();
+                // this condition should never be hit because client side validation is being performed
+                // but just to cover the bases, if this does happen send the erros to the client
+                return Json(new { errors = ModelState.GetErrorMessages() });
             }
 
-            activity.UsersSignedUp.Add(new ActivitySignup
-            {
-                Activity = activity,
-                User = user,
-                SignupDateTime = DateTime.UtcNow
-            });
-
-            await _allReadyDataAccess.UpdateActivity(activity);
-            return new HttpStatusCodeResult((int)HttpStatusCode.OK);
+            _bus.Send(new ActivitySignupCommand() { ActivitySignup = signupModel });
+            return new {Status = "success"};
```

Comment:
Despite the fact that it's not async anymore, I think you should still return an `IActionResult`. Just `return Json(null)` if everything went great.